src/pconn.cc

```diff
@@ -348,33 +350,30 @@ PconnPool::key(const Comm::ConnectionPointer &destLink, const char *domain)
 }
 
 void
-PconnPool::dumpHist(StoreEntry * e) const
+PconnPool::dumpHist(std::ostream &yaml) const
 {
-    storeAppendPrintf(e,
-                      "%s persistent connection counts:\n"
-                      "\n"
-                      "\t Requests\t Connection Count\n"
-                      "\t --------\t ----------------\n",
-                      descr);
+    AtMostOnce title("  persistent connection counts:\n");
```

Comment:
We need to clarify what this report is showing. The output becomes too cryptic if we remove column headings. We should document all the details elsewhere, but we could give more clues here. I suggest something along these lines

```yaml
pool server-peers:
  connection use histogram:
    1: 1
```

or

```yaml
pool server-peers:
  connection use histogram:
    # requests per connection: connections that carried that many requests
    1: 1
```

or

```yaml
pool server-peers:
  connection use histogram:
    # number of requests per connection: number of connections that carried that many requests
    1: 1
```

I only insist on the word histogram (or equivalent).

IIRC, this table is not about persistent connections. It is about all connections, persistent and not. It is about connection "use" rather than just "**re**use".

Reply:
The idea of the comment is very good. I would hoist it one level up, just above the table it refers to (think of the humans)

Reply:
Yes, it is an interesting idea worth considering, but I am not so sure we should add this comment -- the output with a comment is still rather cryptic because it is probably impossible to accurately explain what is going on here in just a few words! If you and @yadij like the comment, let's use it. If there is no consensus, let's not add it (or drop it); I do not insist on (or object to) adding it.

Needless to say, if we do not ban YAML comments here and now, we may find other cases that may benefit from comments, increasing the number of things to fight about. We should keep in mind that report documentation belongs elsewhere. It is difficult to strike the right balance. I am not sure we can come up with some rules of thumb to guide us here if comments are allowed.

Reply:
Let's try to keep it simple.
How about
```
# requests carried per connection: number of connections that carried that many requests
```
It's quite verbose, but should be fairly clear

Reply:
Your call. It would be good to sneak in "closed" (because that caveat is essential for interpreting this statistics, especially in tests with long-lived connections), but I am OK with your wording.

Reply:
done